src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,418 @@
+module Algebra.Graph.Acyclic.AdjacencyMap (
+  -- * Data types and type aliases
+  PartialOrder,
+
+  -- * Data structure
+  AdjacencyMap,
+
+  -- * Internal Checks
+  consistent,
+
+  -- * Graph properties
+  isEmpty, edgeList, vertexList, edgeSet, vertexSet,
+  vertexCount, edgeCount, adjacencyList, hasEdge, hasVertex,
+
+  -- * Graph transformation
+  removeVertex, removeEdge, transpose, induce,
+
+  -- * Basic graph construction primitives
+  empty, vertex, overlayD, connectD, vertices,
+
+  -- * Additional functions on acyclic graphs
+  box, topSort, transitiveClosure,
+
+  -- * Acyclic graph construction methods
+  scc, fromGraph
+  ) where
+
+import Algebra.Graph (Graph, foldg)
+import qualified Algebra.Graph.AdjacencyMap as AM
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AM
+import qualified Algebra.Graph.NonEmpty.AdjacencyMap as NonEmpty
+import qualified Data.Graph.Typed as Typed
+import Data.Set (Set)
+import Data.Coerce (coerce)
+
+{-| The 'AdjacencyMap' data type represents an acyclic graph by
+wrapping around 'Algebra.Graph.AdjacencyMap.AdjacencyMap'. The
+idea is that the user of the library should not be given a way
+to produce a cyclic graph.
+
+We define a 'Num' instance as a convenient notation for working
+with graphs:
+
+  > edgeList   0           == []
+  > vertexList 0           == [0]
+  > edgeList   (1 + 2)     == []
+  > vertexList (1 + 2)     == [1,2]
+  > edgeList   (1 * 2)     == [(1,2)]
+  > vertexList (1 * 2)     == [1,2]
+  > edgeList   (1 + 2 * 3) == [(2,3)]
+  > vertexList (1 + 2 * 3) == [1,2,3]
+  > edgeList   (1 * 2 + 3) == [(1,2)]
+  > vertexList (1 * 2 + 3) == [1,2,3]
+
+The default partial order for the 'Num' instance to restrict
```

Comment:
There is no need to talk about partial orders here -- not everyone knows what they are, and they are not essential for understanding how `Num` instance works. Please simplify this comment. Just say that we only keep edges that are aligned with the order of integers.